host/4/bookworm/python/python312/python312-composite.template

```diff
@@ -0,0 +1,71 @@
+#-------------------------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+#-------------------------------------------------------------------------------------------------------------
+
+FROM mcr.microsoft.com/oryx/python:3.12-debian-bookworm as python
+
+
+# Install Python dependencies
+# MS SQL related packages: unixodbc msodbcsql17 mssql-tools
+# .NET Core dependencies: --no-install-recommends ca-certificates libc6 libgcc1 libgssapi-krb5-2 libicu67 libssl1.1 libstdc++6 zlib1g
+# OpenCV dependencies:libglib2.0-0 libsm6 libxext6 libxrender-dev xvfb
+# binutils: binutils
+# OpenMP dependencies: libgomp1 && \
+# Fix from https://github.com/GoogleCloudPlatform/google-cloud-dotnet-powerpack/issues/22#issuecomment-729895157 : libc-dev
+# mysql dependencies: default-libmysqlclient-dev
+# Azure ML dependencies: liblttng-ust0
+RUN apt-get update && \
+    apt-get install -y wget apt-transport-https curl gnupg locales && \
+    echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections && \
+    curl https://packages.microsoft.com/keys/microsoft.asc | tee /etc/apt/trusted.gpg.d/microsoft.asc && \
```

Comment:
The reason that this location is throwing the error is what I mentioned before. The Oryx image does not have the apt-key for these repositories. I will compare them below to show you what I mean but your team will need to determine how to proceed. Here, the keys are just not available. You could try to move this RUN statement into the final runtime image instead for now. 

![image](https://github.com/Azure/azure-functions-docker/assets/46761504/97480438-a091-4151-8310-b8477536b337)

Error showing that this is the step thats failing as you mentioned.